Add new C function OrtOnnxTypeFromTypeInfo

jignparm opened the pull request:
This is needed to identify if an input or output is a tensor type, or some other type such as sequence or map.

pranavsharma commented:
Please add a unit test as well.

jignparm commented:
> const ONNXType -> ONNXType

Made it "enum ONNXType" instead, to be consistent with OrtGetValueType function.

include/onnxruntime/core/session/onnxruntime_c_api.h:ORT_API(enum ONNXType, OrtGetValueType, _In_ const OrtValue* value)

snnn commented:
> > const ONNXType -> ONNXType
> 
> Made it "enum ONNXType" instead, to be consistent with OrtGetValueType function.
> 
> include/onnxruntime/core/session/onnxruntime_c_api.h:ORT_API(enum ONNXType, OrtGetValueType, _In_ const OrtValue* value)

Did you push your change?

jignparm commented:
Added unit test (i.e. updated an existing unit test) to test this function for Tensor types. For sequence and map types, there are sample models and tests in C#. There's no equivalent model-zoo model to test this function for Sequences or Maps, so excluding that in C-API test for now.

pranavsharma commented on include/onnxruntime/core/session/onnxruntime_c_api.h:
```diff
@@ -358,6 +358,11 @@ ORT_API_STATUS(OrtGetTensorMemSizeInBytesFromTensorProto, _In_ const void* input
+/**
+ * Return OnnxType from OrtTypeInfo
+ */
+ORT_API(enum ONNXType, OrtOnnxTypeFromTypeInfo, _In_ OrtTypeInfo*);
```
can OrtTypeInfo* be a const ptr? Why not?

jignparm replied:
yes, it can/should. Done.
